apps/setting/models_provider/impl/azure_model_provider/model/tts.py

```diff
@@ -0,0 +1,69 @@
+from typing import Dict
+
+from dashscope import api_key
+from dashscope.common.env import api_version
+from openai import OpenAI, AzureOpenAI
+
+from common.config.tokenizer_manage_config import TokenizerManage
+from setting.models_provider.base_model_provider import MaxKBBaseModel
+from setting.models_provider.impl.base_tts import BaseTextToSpeech
+
+
+def custom_get_token_ids(text: str):
+    tokenizer = TokenizerManage.get_tokenizer()
+    return tokenizer.encode(text)
+
+
+class AzureOpenAITextToSpeech(MaxKBBaseModel, BaseTextToSpeech):
+    api_base: str
+    api_key: str
+    api_version: str
+    model: str
+    params: dict
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.api_key = kwargs.get('api_key')
+        self.api_base = kwargs.get('api_base')
+        self.api_version = kwargs.get('api_version')
+        self.model = kwargs.get('model')
+        self.params = kwargs.get('params')
+
+    @staticmethod
+    def new_instance(model_type, model_name, model_credential: Dict[str, object], **model_kwargs):
+        optional_params = {'params': {'voice': 'alloy'}}
+        for key, value in model_kwargs.items():
+            if key not in ['model_id', 'use_local', 'streaming']:
+                optional_params['params'][key] = value
+        return AzureOpenAITextToSpeech(
+            model=model_name,
+            api_base=model_credential.get('api_base'),
+            api_key=model_credential.get('api_key'),
+            api_version=model_credential.get('api_version'),
+            **optional_params,
+        )
+
+    def check_auth(self):
+        client = AzureOpenAI(
+            azure_endpoint=self.api_base,
+            api_key=self.api_key,
+            api_version=self.api_version
+        )
+        response_list = client.models.with_raw_response.list()
+        # print(response_list)
+
+    def text_to_speech(self, text):
+        client = AzureOpenAI(
+            azure_endpoint=self.api_base,
+            api_key=self.api_key,
+            api_version=self.api_version
+        )
+        with client.audio.speech.with_streaming_response.create(
+                model=self.model,
+                input=text,
+                **self.params
+        ) as response:
+            return response.read()
+
+    def is_cache_model(self):
+        return False
```

Comment:
The provided code has several potential issues and areas for optimization:

- **Imports**: The imports of both Dashscope and Azure SDKs should be consistent. If they are different versions, make sure compatibility exists.

```python
# Import only one AI platform here
from openai import OpenAI, AzureOpenAI
```

- **Class Inheritance Issue**: `AzureOpenAITextToSpeech` inherits from `MaxKBBaseModel`, which is not defined in your imported modules. Ensure that `MaxKBBaseModel` and other required models/parent classes are included.

```python
from setting.models_provider.base_model_provider import MaxKBBaseModel
```

- **Missing Initialization Check**: Add checks to initialize variables (`api_key`, `api_base`, etc.) inside the constructor or before using them.

```python
if not self.api_key:
    raise ValueError("API Key cannot be empty")

if api_url == "":
    raise ValueError("API URL cannot be empty")
```

- **Static Method Usage`: Static methods should use their class name instead of `cls`. 

```python
@staticmethod
def new_instance(class_name, model_name, model_credential: Dict[str, object], **model_kwargs):
```

- **Parameters Handling**: Simplify how parameters are handled in the static method to reduce redundancy.

```python
@staticmethod
def new_instance(model_type, model_name, model_credentials, **model_kwargs):
    optional_params = {'params': {'voice': 'alloy'}}
    final_params = {**optional_params, **{k: v for k,v in model_kwargs.items() if k != "use_local"}}
    return cls(
        model=final_params['model'],
        api_base=model_credentials.get('api_base'),
        api_key=model_credentials.get('api_key'),
        api_version=modesl_credentials.get('api_version'),
        **final_params
    )
```

- **Simplify Text-to-Speech Functionality**: Use built-in functionalities provided by OpenAI/Azure if available, possibly reducing boilerplate.

```python
def text_to_speech(self, text):
    client = AzureOpenAI(
        azure_endpoint=self.api_base,
        api_key=self.api_key,
        api_version=self.api_version
    )

    # Instead of creating a streaming response explicitly and reading data, 
    # we can stream directly from the response generator
    voices = client.voice.all()

    try:
        voice = next(voices)
        result = client.audio.speech.create(
            model=self.model,
            gender=voice.gender,
            text=text,
            **self.params
        )
        
        audio_data = b""
        async for chunk in result.streaming_audio():
            audio_data += bytes(chunk)

        return audio_data

    except StopIteration:
        raise Exception("No matching voice found.")
``` 

These adjustments help ensure cleaner and more efficient code while addressing logical errors related to API authentication and handling specific functionalities. Note that error checking like this helps improve robustness during runtime execution.